File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n - lost.length;
        HashSet<Integer> hs = new HashSet<>();
        Arrays.sort(lost);
        
        for (int i : reserve){
            hs.add(i);
        }
        
        for (int i = 0; i < lost.length; i++){
            if (hs.contains(lost[i])){
                answer += 1;
                hs.remove(lost[i]);
                lost[i] = 0;
            }
        }
        
        for (int i : lost){
            if (i != 0){
                if (hs.contains(i - 1)){
                    answer += 1;
                    hs.remove(i - 1);
                }else if (hs.contains(i + 1)){
                    answer += 1;
                    hs.remove(i + 1);
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = n - len(lost)
    lost.sort()
    reserve.sort()
    
    for i in reserve[:]:
        if i in lost:
            answer += 1
            lost.remove(i)
            reserve.remove(i)
            
    for i in reserve:
        if i-1 in lost:
            lost.remove(i-1)
            answer += 1
            
        elif i+1 in lost:
            lost.remove(i+1)
            answer += 1
        
    return answer